elytron/WFLY-10810_Elytron_JASPI_Servlet_Profile_Support.adoc

```diff
@@ -0,0 +1,160 @@
+
+Each of these four areas will now be explored further.
+
+==== Concrete AuthConfigFactory implementation and registration.
+
+The class `javax.security.auth.message.config.AuthConfigFactory` WildFly Elytron will provide a concrete implementation of this class, there is nothing remarkable about this class other than it is required to implement the API as specified.
+
+The WildFly Elytron subsystem will have a new attribute `register-jaspi-factory` to control if the subsystem should instantiate and register the new factory, instantiation and registration of the factory will happen if the following two conditions are met: -
+. `register-jaspi-factory` is set to `true` (It's default value).
+. No `authconfigprovider.factory` system property has been set specifying an alternative implementation is required
+
+Upon registering the `AuthConfigFactory` the subsystem will first call `AuthConfigFactory.getFactory()` to identify if the PicketBox implementation of the factory is also available, if it is the two factories (Elytron and PicketBox) will be wrapped in a delegating factory which by default will delegate to the Elytron factory but will delegate to the PicketBox factory if the following two conditions are met: -
+. The request configuration was not available from the Elytron factory.
+. The current deployment accessing the factory is secured using PicketBox.
+
+The reason priority is given to configurations from the Elytron factory is these may have been programatically registered at a later point and must take higher priority.  
+
+The reason a deployment must be using PicketBox security before delegation is because the PicketBox implementation has default 'catch all' configurations which we do not wish to be triggered when not using PicketBox security. 
+
+==== AuthConfigProvider implementation and configuration support.
+
+The JASPI specification defines a very flexible API which allows for programatic manipulation by deployments, however it does not provide any API for actually instantiating an `AuthConfigProvider` or related classes to be able to register them.
+
+This RFE is focused on support for the Servlet profile which defines server side request handling only, WildFly Elytron implementations of `javax.security.auth.message.config.AuthConfigProvider`, `javax.security.auth.message.config.ServerAuthConfig`, and `javax.security.auth.message.config.ServerAuthContext`.  Again these will be implemented according to the requirements of the API however we are given an opportunity to define our own behaviour in relation to how we support multiple authentication modules.
+
+Our `ServerAuthConfig` implementation will include the following features: -
+* The `ServerAuthModule` instances will be strictly ordered.
+* Each will be associated with a flag to control the behaviour in the event of success / failure (The result of these flags will be specified in the documentation)
+** Required
+** Requisite
+** SUFFICIENT
```

Comment:
Are these two in upper case intentionally?

Reply:
No I will correct that tomorrow, this is only supposed to be an illustration of the flags that will be supported not specify an exact API.